OpenRA.Mods.Common/Traits/Player/ProductionQueue.cs

```diff
@@ -101,13 +104,28 @@ public ProductionQueue(ActorInitializer init, Actor playerActor, ProductionQueue
 			developerMode = playerActor.Trait<DeveloperMode>();
 
 			Faction = init.Contains<FactionInit>() ? init.Get<FactionInit, string>() : self.Owner.Faction.InternalName;
-			Enabled = !info.Factions.Any() || info.Factions.Contains(Faction);
+			UsableByFaction = !info.Factions.Any() || info.Factions.Contains(Faction);
+			ProductionAvailable = UsableByFaction;
```

Comment:
This will be replaced by `allCompatibleProductionTraitPairs.Any()` in `INotifyCreated`.  I guess you want to remove this, and change that to `UsableByFaction && allCompatibleProductionTraitPairs.Any()`?